Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java

```diff
@@ -14,14 +16,22 @@
  */
 public class PayMoreToCastAsThoughtItHadFlashAbility extends SpellAbility {
 
-    private final ManaCosts costsToAdd;
+    private final Cost costsToAdd;
 
     public PayMoreToCastAsThoughtItHadFlashAbility(Card card, ManaCosts<ManaCost> costsToAdd) {
-        super(card.getSpellAbility().getManaCosts().copy(), card.getName() + " as though it had flash", Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
+        this(card, (Cost) costsToAdd);
+        CardUtil.increaseCost(this, costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Costs<Cost> costsToAdd) {
```

Comment:
so you added two new constructors, because the cost increase can't be handled in the general case?

manacosts need ` CardUtil.increaseCost` while other costs need `this.addCost`

is it really necessary to have separate constructors, or can you just use a single general constructor and check `instanceof ManaCosts` to determine how to do it?

Reply:
No, it's really not necessary. I was only afraid of uncheked cast warning (`Cost` to `ManaCosts<ManaCost>`), but after your review I examined hierarchy and noticed that there's no problem. I've changed it.
It looks much clearer now! Thank you!